src/main/java/edu/harvard/iq/dataverse/util/MailSessionProducer.java

```diff
@@ -0,0 +1,132 @@
+            Context initialContext = new InitialContext();
+            this.appserverProvidedSession = (Session)initialContext.lookup("mail/notifyMailSession");
+        } catch (NamingException e) {
+            // This exception simply means the appserver did not provide the legacy mail session.
+            // Debug level output is just fine.
+            logger.log(Level.FINE, "Error during mail resource lookup", e);
+        }
+    }
+    
+    @Produces
+    @Named("mail/systemSession")
+    public Session getSession() {
+        //  For backward compatibility, prefer to return the mail resource configured on the appserver.
+        if (appserverProvidedSession != null) {
+            logger.warning("The configuration of mail transfer agents using asadmin create-javamail-resource is" +
+                " deprecated. Please migrate to using JVM options, see Dataverse guides for details");
+            return appserverProvidedSession;
+        }
+        
+        if (systemMailSession == null) {
+            // Initialize with null (= no authenticator) is a valid argument for the session factory method.
+            Authenticator authenticator = null;
+            
+            // In case we want auth, create an authenticator (default = false from microprofile-config.properties)
+            if (JvmSettings.MAIL_MTA_AUTH.lookup(Boolean.class)) {
+                authenticator = new Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(JvmSettings.MAIL_MTA_USER.lookup(), JvmSettings.MAIL_MTA_PASSWORD.lookup());
+                    }
+                };
+            }
+            
+            this.systemMailSession = Session.getInstance(getMailProperties(), authenticator);
+        }
+        return systemMailSession;
+    }
+    
+    Properties getMailProperties() {
+        Properties configuration = new Properties();
+        
+        // See https://jakarta.ee/specifications/mail/2.1/apidocs/jakarta.mail/jakarta/mail/package-summary
+        configuration.put("mail.transport.protocol", "smtp");
+        configuration.put("mail.debug", JvmSettings.MAIL_DEBUG.lookupOptional(Boolean.class).orElse(false).toString());
+        // Only enable if your MTA properly supports UTF-8 mail addresses following RFC 6530/6531/6532.
+        // Before, we used a hack to put the raw UTF-8 mail address into the system.
+        // Now, make it proper, but make it possible to disable it - see also EMailValidator.
+        // Default = true from microprofile-config.properties as most MTAs these days support SMTPUTF8 extension
+        configuration.put("mail.mime.allowutf8", JvmSettings.MAIL_MTA_SUPPORT_UTF8.lookup(Boolean.class).toString());
+        
+        // Map properties 1:1 to mail.smtp properties for the mail session.
+        smtpStringProps.forEach(
```

Comment:
Are type specific iterators needed? If I'm not mistaken jakarta mail only uses `string` to encode values.

Reply:
Yes, they are needed here, so we can provide minimal input validation. Yes, Jakarta Mail will accept all strings, but IMHO it is useful to support a user when we do have proper type information.